### PR TITLE
MAGECLOUD-3205: Provide ElasticsSuite auto-configuration in ECE-Tools

### PR DESCRIPTION
**Pull request by shiftedreality**
## Scope of this PR

- __Major Update__: Significant new info: new section in existing topic, etc.

## Purpose of this PR

https://magento2.atlassian.net/browse/MAGECLOUD-3205

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/cloud/env/variables-deploy.md [hunk 1]
```diff
@@ -398,6 +398,45 @@ stage:
...
+        number_of_replicas: 3
+      _merge: true
+```
+
+#### Known limitations
+There are some known issues you should be aware of once `ElasticSuite` module is installed:
+
+* Changing `search engine` to anything unlike `elasticsuite` would cause failed deploy with appropriate validation error
+* Removing `ElasticSearch` service would cause failed deploy with appropriate validation error
+
```
Comment (NadiyaS) [inline guides/v2.2/cloud/env/variables-deploy.md]: Should not we mention somewhere how to add ElasticSuite with link to its documentation? and once you added this package to the project - your search engine is changed to elasticsuite automatically (and on cloud) and you should uninstall elasticsuite if wants to switch to another engine

Comment (hguthrie) [reply guides/v2.2/cloud/env/variables-deploy.md]: DevDocs have been told to state specifically that we do not support ElasticSuite. Has this changed?

Comment (hguthrie) [reply guides/v2.2/cloud/env/variables-deploy.md]: See here we even call it out: https://devdocs.magento.com/guides/v2.2/cloud/project/project-conf-files_services-elastic.html#add-elasticsearch-plugins

Please let me know if that has changed

Comment (shiftedreality) [reply guides/v2.2/cloud/env/variables-deploy.md]: Yes, we added functionality to correctly set up this module

Comment (jeff-matthews) [reply guides/v2.2/cloud/env/variables-deploy.md]: _**If**_ Magento's support position on Elasticsuite _**has**_ changed, we'll also need an update here: https://devdocs.magento.com/guides/v2.3/cloud/project/project-conf-files_services-elastic.html#elasticsearch-plugins.

Comment (shiftedreality) [reply guides/v2.2/cloud/env/variables-deploy.md]: Maybe we can call it *limited support*, need advise from @YPyltiai

Comment (jeff-matthews) [reply guides/v2.2/cloud/env/variables-deploy.md]: Seems like we should also run this by someone in Support.

Comment (YPyltiai) [reply guides/v2.2/cloud/env/variables-deploy.md]: We **do not** Support ElasticSuite module itself as well as issues in it. We **do** configure it properly for Cloud, though which is the only thing we "Support". Being short - as long as the configuration of module and connection to ES servers are set up correctly - our support for this module ends at this point.

**Pull request merged**